packages/peregrine/lib/talons/Header/useStoreSwitcher.js

```diff
@@ -80,16 +80,53 @@ export const useStoreSwitcher = props => {
         // Change store view code and currency to be used in Apollo link request headers
         storeCode => {
             // Do nothing when store view is not present in available stores
-            if (!availableStores.get(storeCode)) return;
+            if (!availableStores.has(storeCode)) return;
 
             storage.setItem('store_view_code', storeCode);
             storage.setItem(
                 'store_view_currency',
                 availableStores.get(storeCode).currency
             );
 
-            // Refresh the page to re-trigger the queries once code/currency are saved in local storage.
-            history.go(0);
+            // Handle updating the URL if the store code should be present.
+            // In this block we use `window.location.assign` to work around the
+            // static React Router basename, which is changed on initialization.
+            if (process.env.USE_STORE_CODE_IN_URL === 'true') {
+                const pathName = window.location.pathname;
+                const params = window.location.search;
+
+                // Check to see if we're on a page outside of the homepage
+                if (pathName !== '' && pathName !== '/') {
+                    const [, pathStoreCode] = pathName.split('/');
+
+                    // If the current store code is in the url, replace it with
+                    // the new one.
+                    if (
+                        availableStores.has(pathStoreCode) &&
+                        availableStores.get(pathStoreCode).isCurrent
+                    ) {
+                        const newPath = `${pathName.replace(
+                            `/${pathStoreCode}`,
+                            `/${storeCode}`
+                        )}${params}`;
+
+                        window.location.assign(newPath);
+                    } else {
+                        // Otherwise include it and reload.
+                        const newPath = `/${storeCode}${
```

Comment:
I would suggest you move the `pathName && pathName !== '/' ? pathName : ''` check outside of the string template for ease of reading.

Reply:
Removed the conditional entirely since it is assumed from the wrapping condition anyways.